src/routes/github/subscription/github-subscription-delete.ts

```diff
@@ -2,26 +2,38 @@ import { Request, Response } from "express";
 import { Subscription } from "models/subscription";
 import { isUserAdminOfOrganization } from "~/src/util/github-utils";
 import { createAppClient, createUserClient } from "~/src/util/get-github-client-config";
+import { GitHubServerApp } from "models/github-server-app";
 
 export const GithubSubscriptionDelete = async (req: Request, res: Response): Promise<void> => {
-	const { githubToken, jiraHost, gitHubAppId } = res.locals;
+	const { githubToken, jiraHost, gitHubAppConfig } = res.locals;
 	const { installationId: gitHubInstallationId } = req.body;
 	const logger = req.log.child({ jiraHost, gitHubInstallationId });
+
+	logger.debug("Received DELETE subscription request");
+
+	const gitHubApp = await GitHubServerApp.findForUuid(req.params.uuid);
+
+	if (gitHubApp?.id !== gitHubAppConfig?.gitHubAppId) {
+		logger.debug("GitHub app IDs do not match. Cannot DELETE subscription.");
+		throw new Error("Cannot DELETE subscription.");
```

Comment:
Extra precaution to make sure we're deleting the correct gh app and that the user has permissions to do so.